pkg/setting/date_formats.go

```diff
@@ -0,0 +1,28 @@
+package setting
+
+type DateFormats struct {
+	FullDate         string              `json:"fullDate"`
+	UseBrowserLocale bool                `json:"useBrowserLocale"`
+	Interval         DateFormatIntervals `json:"interval"`
+}
+
+type DateFormatIntervals struct {
+	Second string `json:"second"`
+	Minute string `json:"minute"`
+	Hour   string `json:"hour"`
+	Day    string `json:"day"`
+	Month  string `json:"month"`
+	Year   string `json:"year"`
+}
+
+func (cfg *Cfg) readDateFormats() {
+	preferences := cfg.Raw.Section("date_formats")
+	cfg.DateFormats.FullDate, _ = valueAsString(preferences, "full_date", "YYYY-MM-DD HH:mm:ss")
+	cfg.DateFormats.Interval.Second, _ = valueAsString(preferences, "interval_second", "HH:mm:ss")
+	cfg.DateFormats.Interval.Minute, _ = valueAsString(preferences, "interval_minute", "HH:mm")
+	cfg.DateFormats.Interval.Hour, _ = valueAsString(preferences, "interval_hour", "MM-DD HH:mm")
+	cfg.DateFormats.Interval.Day, _ = valueAsString(preferences, "interval_day", "YYYY-MM-DD")
+	cfg.DateFormats.Interval.Month, _ = valueAsString(preferences, "interval_month", "YYYY-MM")
```

Comment:
Need to handle possible returned error values

Reply:
This cannot error, we do not do that in many places

Reply:
valueAsString function looks pretty bad, and the panic recover? why not just check if key exists? settings.go needs a lot of refactoring, cannot have error handling duplicated for every key that seems really  bad